server/service/oauth.go

```diff
@@ -88,12 +98,83 @@ func (o *OAuth) GetUser(ctx context.Context, code string) (*model.User, error) {
 	var user githubUser
 	err = json.NewDecoder(resp.Body).Decode(&user)
 	if err != nil {
-		return nil, fmt.Errorf("can't parse github response: %s", err)
+		return nil, fmt.Errorf("can't parse github user response: %s", err)
+	}
+
+	if o.restrictAccess != "" {
+		if err := o.checkAccess(client, o.restrictAccess, user.Login); err != nil {
+			return nil, err
+		}
 	}
+
+	role := model.Requester
+	if o.restrictReviewerAccess != "" {
+		err := o.checkAccess(client, o.restrictReviewerAccess, user.Login)
+		if err != nil && err != NoAccessError {
+			return nil, err
+		}
+		if err == NoAccessError {
+			role = model.Worker
+		}
+	}
```

Comment:
With the code above, the reviewers are a subgroup of the workers, right?

This is not wrong, but it does not match with what we discussed IRL. We said that for greater flexibility we would allow independent orgs/teams for each role.

What about something like this?
```
	role := model.Requester

	if o.restrictReviewerAccess != "" {
		err := o.checkAccess(client, o.restrictReviewerAccess, user.Login)
		if err != nil && err != NoAccessError {
			return nil, err
		}
		if err == NoAccessError {
			role = model.Worker
		}
	}

	if role == model.Worker && o.restrictAccess != "" {
		if err := o.checkAccess(client, o.restrictAccess, user.Login); err != nil {
			return nil, err
		}
	}
```